dev/tools/localization/gen_l10n.dart

```diff
@@ -473,13 +470,109 @@ Future<void> main(List<String> arguments) async {
     'pub',
     'run',
     'intl_translation:generate_from_arb',
-    '--output-dir=${l10nDirectory.path}',
+    '--output-dir=${localizationsGenerator.l10nDirectory.path}',
     '--no-use-deferred-loading',
-    outputFile.path,
+    localizationsGenerator.outputFile.path,
     ...arbFilenames,
   ]);
   if (generateFromArbResult.exitCode != 0) {
     stderr.write(generateFromArbResult.stderr);
     exit(1);
   }
 }
+
+/// The localizations generation class used to generate the localizations
+/// classes, as well as all pertinent Dart files required to internationalize a
+/// Flutter application.
+class LocalizationsGenerator {
```

Comment:
It doesn't really look like this class is doing localization generation, just collecting some directories for processing.